component/remote/http/http.go

```diff
@@ -0,0 +1,269 @@
+// Package http implements the remote.http component.
+package http
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+	common_config "github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/pkg/build"
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+	"github.com/grafana/agent/pkg/river"
+	prom_config "github.com/prometheus/common/config"
+)
+
+var userAgent = fmt.Sprintf("GrafanaAgent/%s", build.Version)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "remote.http",
+		Args:    Arguments{},
+		Exports: Exports{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments control the remote.http component.
+type Arguments struct {
+	URL           string        `river:"url,attr"`
+	PollFrequency time.Duration `river:"poll_frequency,attr,optional"`
+	PollTimeout   time.Duration `river:"poll_timeout,attr,optional"`
+	IsSecret      bool          `river:"is_secret,attr,optional"`
+
+	Client common_config.HTTPClientConfig `river:"client,block,optional"`
+}
+
+// DefaultArguments holds default settings for Arguments.
+var DefaultArguments = Arguments{
+	PollFrequency: 1 * time.Minute,
+	PollTimeout:   10 * time.Second,
+	Client:        common_config.DefaultHTTPClientConfig,
+}
+
+var _ river.Unmarshaler = (*Arguments)(nil)
+
+// UnmarshalRiver implements river.Unmarshaler.
+func (args *Arguments) UnmarshalRiver(f func(interface{}) error) error {
+	type arguments Arguments
+	if err := f((*arguments)(args)); err != nil {
+		return err
+	}
+
+	if args.PollFrequency == 0 {
+		return fmt.Errorf("poll_frequency must be greater than 0")
```

Comment:
Should we check for negative durations too? They're valid River expressions, too. 
(This is true for some other components that use durations like GCFrequency on otelcol.exporter.prometheus)